Lib/os.py

```diff
@@ -697,7 +697,8 @@ def __delitem__(self, key):
             raise KeyError(key) from None
 
     def __iter__(self):
-        for key in self._data:
+        data = list(self._data)
```

Comment:
Either inline `data` or rename it to `keys`. Add an explaining comment about atomicity of list() from dict.